File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(start, end):
    result = []
    for i in range(start, end):
        sum = 0
        num = i
        digits = str(num)
        digits = digits.split()
        for j in range(0, len(digits)):
            #power = digits[j] ** len(digits)
            power = pow(int(digits[j]), len(digits))
            sum += power
            if sum == num:
                result.append(num)
    print(result)

print(find_armstrong_numbers(0,999))

File: js/armstrongNumbers.js
function armstrong(start, end) {
    const result = [];
    for (let i = start; i <= end; i++){
      let sum = 0;
      const num = i; // just to make it easier to understand
      //split number into digits
      const digits = num.toString().split('');
      //iterate through digits
      for (let j = 0; j < digits.length; j++){
        // calculate number to powwer
        digits[j] = Math.pow(digits[j], digits.length)
        sum += digits[j]
        if (sum == num) {
          result.push(sum)
        }
      } 
    }
    return result;
  }
  
  console.log(armstrong(0,999))